File: e2e/selenide-junit5-selenium-grid/src/test/java/provider/SupportProvider.java
package provider;

import lombok.Getter;
import support.api.ApiClient;
import support.mock.ApiRouteMockClient;
import support.mock.WireMockClient;

@Getter
public class SupportProvider {
    public ApiClient api = new ApiClient();
    public WireMockClient wiremock = new WireMockClient();
    public ApiRouteMockClient mock = new ApiRouteMockClient();

    public void clearMocks() {
        mock.reset();
    }
}

File: e2e/maestro/providers/SupportProvider.ts
import { AllureClient } from '@/support/allure/AllureClient';
import { ApiRouteMock } from '@/support/mocks/ApiRouteMock';
import { WireMockClient } from '@/support/mocks/WireMockClient';
import type { TestSuite } from '@/runner/types';

export class SupportProvider {
  readonly wiremock: WireMockClient;
  readonly mock: { api: ApiRouteMock };
  readonly allure: AllureClient;

  constructor(resultsDir: string, suite: TestSuite) {
    this.wiremock = new WireMockClient();
    this.mock = { api: new ApiRouteMock(this.wiremock) };
    this.allure = new AllureClient(resultsDir, suite);
  }

  async reset(): Promise<void> {
    await this.wiremock.reset();
    await this.mock.api.getTasks();
  }
}
